R/missingness.R

```diff
@@ -134,3 +134,60 @@ countMissingData <- function(x, userNAs = NULL) {
   .Deprecated("missingness", "healthcareai")
   missingness(x)
 }
+
+#' Summarizes data given by \code{\link{missingness}}
+#'
+#' @description Interpreting \code{\link{missingness}} results from wide
+#'   datasets is difficult. This function helps interpret missingness output by
+#'   summarizing this output by listing: the percent of variables that
+#'   contain missingness, the variable name of the variable with the maximum
+#'   amount of missingness along with its percent missingness, and a tibble
+#'   that lists the top 5 missingness levels with the count of the number of
+#'   variables associated with each level.
+#' @param object Data frame from \code{\link{missingness}}
+#' @param ... Unused
+#' @return a tibble of the top 5 missingness percentage levels with the count of
+#'   the number of variables associated with each level
+#' @export
+#' @examples
+#' missingness(pima_diabetes) %>%
+#'   summary()
+#'
+summary.missingness <- function(object, ...) {
+  if (!length(object))
+    stop("`object` is empty.")
+
+  col_missing <- pull(object, percent_missing) > 0
+  if (sum(col_missing) == 0) {
+    stop("`object` has no variables with missingness.")
+  } else {
+    # Get the percent of columns that have any missing values
+    perc_col_missing <- mean(col_missing) * 100 # Convert from decimal to percent
+
+    # Get the name and percent_missing of the variable with the most missingness
+    max_df <- (
+      object %>% filter(percent_missing == max(percent_missing))
+    )[1, ] #In tie, grab first row
+
+    out <- paste0("Missingness summary:\n", perc_col_missing,
+                  "% of data variables contain missingness.\n`",
+                  max_df$variable,
+                  "` contains the most missingness with ",
+                  round(max_df$percent_missing, 1),
+                  "% missingness.\n\nNumber of variables with levels of ",
```

Comment:
Let's try to provide more intuition instead of repeating missingness on this line. What do you think of "x contains the most missingness with y% of observations containing missing values"?

Reply:
That is better. I like that.